src/core/amount.js

```diff
@@ -729,7 +705,9 @@ Amount.prototype.parse_native = function(j) {
 // Requires _currency to be set!
 Amount.prototype.parse_value = function(j) {
   this._is_native = false;
-  this._set_value(new BigNumber(j), BigNumber.ROUND_DOWN);
+  const newValue = new IOUValue(
+    new IOUValue(j).toPrecision(16, Value.getBNRoundDown()));
```

Comment:
It seems messier to me to be passing another value into the constructor for one specific case versus calling a specific function. But I can change it